### PR TITLE
PIP-1440-madQC

### PR DESCRIPTION
**Pull request by ottojolanki**
Add logic to post madQC on unreplicated experiments that have 2 technical replicates.

### CODE REVIEW

File: accession/accession.py [hunk 1]
```diff
@@ -1041,6 +1032,25 @@ def prepare_mad_qc_metric(self, gs_file: GSFile) -> Dict[str, Any]:
...
+        elif self.experiment.get_number_of_biological_replicates() > 2:
+            return
+        elif (
+            self.experiment.get_number_of_biological_replicates() == 1
+            and self.experiment.get_number_of_technical_replicates() == 2
+        ) or self.experiment.get_number_of_biological_replicates() == 2:
+            mad_qc = self.prepare_mad_qc_metric(gs_file)
+            return self.queue_qc(mad_qc, encode_file, "mad-quality-metric", shared=True)
+        else:
+            return
```
Comment (paul-sud) [inline accession/accession.py]: `self.experiment.get_number_of_biological_replicates()` is called three times. Would be good to consolidate:
```suggestion
        if encode_file.has_qc("MadQualityMetric"):
            return
           
        num_biological_replicates = self.experiment.get_number_of_biological_replicates()
        if num_biological_replicates > 2:
            return
        elif (
            num_biological_replicates == 1
            and self.experiment.get_number_of_technical_replicates() == 2
        ) or num_biological_replicates == 2:
            mad_qc = self.prepare_mad_qc_metric(gs_file)
            return self.queue_qc(mad_qc, encode_file, "mad-quality-metric", shared=True)
        else:
            return
```

**Review by paul-sud — CHANGES_REQUESTED** [review]
One comment, otherwise looks good.

File: accession/accession.py (hunk 1 above)
Comment (ottojolanki) [reply accession/accession.py]: Yeah, considered this and decided that three dict accesses is better than extra variable :) I'm ok either way.

**Comment by paul-sud** [comment]
@ottojolanki could you lint and re-push? Will merge once tests pass.

**Pull request merged**